### ISSUE TITLE
Various issues in 5.2-develop

### ISSUE THREAD

**Issue by lusum** [ampache/ampache#3135]
### Describe the bug

Some issues in 5.2-develop

1. ampache_cron.timer and channel_restart.timer have a bug

in install section

'[Install]
'WantedBy=timer.target

the correct code is:

'WantedBy=timer.starget

(see ampache_podcasts.timer  )

2. when you create a smart playlist, the playlist count in api is not updated unless you update the catalog

3. Annoying with large catalogs: if you update a catalog in web interface ( cli is ok ) , it seems to trigger optimisation of every other catalog ( btw: optimisation should be optional also in web interface )

**Comment by lachlan-00** [comment]
all done bar the db optimization now.

1d2d2925d
6b10217d6

**Comment by lachlan-00** [comment]
looks like a malformed url
```
/admin/catalog.php?action=add_to_catalog&catalogs[]=21
```

**Comment by lusum** [comment]
Db optimisation has another problem: if you do multiple operations on the catalog ( add/delete ) in the web interface, optimisation starts every time and, sometime you have multiple optimisation running in parallel. Now, for big catalogs, it could be a big issue, maybe should be better to add catalog optimisation as a separate menu item

**Comment by lachlan-00** [comment]
probably need to do a lock of some kind to stop that repeating itself

**Comment by lachlan-00** [comment]
probably need to do a lock of some kind to stop that repeating itself

**Comment by lusum** [comment]
If it could be possible, optimisation should be a low priority task

**Comment by lachlan-00** [comment]
i've set it to ignore the whole process if it happens within 30mins of the last event.

Might make it longer though, or make some of the queries wait even longer than that as they're not common issues with catalog changes

**Comment by lusum** [comment]
It is possible to separate optimisation from other actions in catalog web interface? For large catalogs it  takes a lot of time and the speed improvement is negligible.

**Comment by lusum** [comment]
For problem 4  (optimisation ) : it should be possible to avoid automatic optimisation using an option in config file

For problem 3 ( playlist count not updated ) : two possible solutions: first one, calling User::update_counts() after playlist (o r smart playlist creation of deletion )
second one: call a function similar to User::update_counts() but call it User::update_counts only for a specific kind ( playlist, live_streams )

Problem 3 affects also live_stream and maybe podcasts.

**Comment by lusum** [comment]
It seems ok now
